tests/test_dockerfile.py

```diff
@@ -128,3 +125,53 @@ def test_volumes(self):
             dockerfile = f.read()
             regex = re.compile(r'.*VOLUME \["/var/lib"\]\nVOLUME \["/usr/lib"\]',  re.MULTILINE)
             self.assertRegexpMatches(dockerfile, regex)
+
+    def test_default_substitution(self):
+        with open(self.yaml, 'ab') as f:
+            f.write("from: {{FROM:rhel:7}}\nversion: 1.0".encode())
+
+        generator = Generator(self.log, self.args)
+        generator.configure()
+        generator.render_from_template()
```

Comment:
this code is copy pasted multiple times maybe it should be extracted to common setUp method